images/haproxy/Dockerfile

```diff
@@ -13,6 +13,6 @@
 # limitations under the License.
 
 # standard haproxy image + minimal config so the container will not exit
-ARG BASE="haproxy:2.2.0-alpine"
```

Comment:
what's the size difference?
we just need an image with haproxy IIRC (and maybe `cat` ?). we could also do some COPY / multi-stage onto distroless or something

Reply:
```
haproxy         2.2.0                d27dbbd0e8a9   11 months ago   93.3MB
haproxy         2.2.0-alpine         0b8abcf7bfdb   12 months ago   24.5MB
```

Reply:
slightly unfortunate, but this image is also the one we're least sensitive to.

I think we really just need:
- haproxy
- kill
- cat

It's too bad there isn't a base image closer to just haproxy.

Reply:
@BenTheElder so start with the ubuntu image and COPY to distroless with those 3 binaries? is there a optional presubmit CI job that would be useful in testing a change this?

Reply:
I don't think we have one currently, there's not been much demand for the HA functionality.